pootle/static/js/editor/components/SuggestionFeedbackForm.js

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) Pootle contributors.
+ *
+ * This file is a part of the Pootle project. It is distributed under the GPL3
+ * or later license. See the LICENSE file for a copy of the license and the
+ * AUTHORS file for copyright and authorship information.
+ */
+
+import _ from 'underscore';
+
+import React from 'react';
+
+import FormElement from 'components/FormElement';
+import FormMixin from 'mixins/FormMixin';
+
+export const SuggestionFeedBackForm = React.createClass({
+
+  propTypes: {
+    suggId: React.PropTypes.number.isRequired,
+    initialSuggestionText: React.PropTypes.string.isRequired,
+    localeDir: React.PropTypes.string.isRequired,
+    onAcceptSuggestion: React.PropTypes.func.isRequired,
+    onRejectSuggestion: React.PropTypes.func.isRequired,
+    onChange: React.PropTypes.func.isRequired,
+  },
+
+  mixins: [FormMixin],
+
+  /* Lifecycle */
+
+  getInitialState() {
+    this.initialData = {
+      comment: '',
+      translation: this.props.initialSuggestionText,
+    };
+
+    return {
+      formData: this.initialData,
+    };
+  },
+
+  fields: ['comment'],
+
+  /* Handlers */
+
+  handleAccept(e) {
+    const suggestionChanged = (
+      this.state.formData.translation !== this.props.initialSuggestionText
+    );
+    e.preventDefault();
+    this.props.onAcceptSuggestion(
+      this.props.suggId,
+      {
+        requestData: this.state.formData,
+        isSuggestionChanged: suggestionChanged,
+      }
+    );
+  },
+
+  handleReject(e) {
+    e.preventDefault();
+    this.props.onRejectSuggestion(this.props.suggId, { requestData: this.state.formData });
+  },
+
+  handleStateChange(name, value) {
+    const newData = _.extend({}, this.state.formData);
+    newData[name] = value;
+    const isDirty = !_.isEqual(newData, this.initialData);
+    this.setState({ isDirty, formData: newData });
+    this.props.onChange(isDirty);
+  },
+
+  /* Layout */
+
+  render() {
+    const { errors } = this.state;
+    const { formData } = this.state;
+    const autoFocus = true;
```

Comment:
You probs misunderstood. But I'll check again.
I had the error when I used autoFocus={true} in the JSX element.